ARCHITECTURE.md

```diff
@@ -6,6 +6,233 @@ This document describes the high-level architecture of Ditto Network JS SDK.
 > HERE IS THE VERSION OF THE ARCHITECTURE THAT WAS DESIGNED. HOWEVER, THE ACTUAL IMPLEMENTATION ARE DIFFERENT. SO WE'LL JUST SAVE IT FOR A WHILE AND SYNCHRONIZE IT WITH THE REAL IMPLEMENTATION LATER.
 
 
+## Workflow Configuration Specification
+
+JSON serializable format
+
+- **issuer**: (String) The entity responsible for the workflow.
+- **name**: (String) A brief and descriptive name for the workflow.
+- **hint**: (String) A short description or hint about what the workflow does.
+- **triggers**: (Array) A list of trigger objects that define when the workflow should be executed.
+- **actions**: (Array) A list of action objects that define what operations the workflow performs.
+
+```json
+{
+  "issuer": "Ditto",
+  "name": "Send assets to multiple addresses",
+  "hint": "Send your assets to multiple addresses in one transaction",
+  "triggers": [
+    {
+      "type": "Core.Scheduled",
+      "options": {
+        "frequency": "daily",
+        "time": "08:00"
+      }
+    }
+  ],
+  "actions": [
+    {
+      "type": "Application.MultiSender",
+      "options": {
+        "items": [
+          {
+            "asset": {
+              "address": "0xTokenAddress",
+              "decimals": 18
+            },
+            "amount": "100",
+            "to": "0xRecipientAddress"
+          },
+          {
+            "asset": null,
+            "amount": "0.1",
+            "to": "0xAnotherRecipientAddress"
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+
+### Trigger and Action Objects
+
+#### Trigger
+
+A trigger object defines the conditions under which the workflow is executed.
+
+- **type**: (String) The type of trigger. This should correspond to a predefined trigger type in the application.
+- **options**: (Object) An optional set of parameters specific to the trigger type.
+
+```json
+{
+  "type": "Core.Scheduled",
+  "options": {
+    "frequency": "daily",
+    "time": "08:00"
+  }
+}
+```
+
+- **type**: Specifies a scheduled trigger.
+- **options**:
+  - **frequency**: The frequency with which the trigger should activate (e.g., "daily", "weekly").
+  - **time**: The specific time of day the trigger should activate.
+
+
+#### Action
+
+An action object defines the operations that are performed when the workflow is triggered.
+
+- **type**: (String) The type of action. This should correspond to a predefined action type in the application.
+- **options**: (Object) An optional set of parameters specific to the action type.
+
+
+```json
+{
+  "type": "Application.MultiSender",
```

Comment:
Should we include the version?
For example we change the input parameters for an action, and user makes a call from the old version of sdk
Then it will fail deserializing on our side

Reply:
I think it will be configured via package version itself